frame/lottery/src/lib.rs

```diff
@@ -442,12 +455,12 @@ impl<T: Config> Pallet<T> {
 						Error::<T>::AlreadyParticipating
 					);
 				}
+				participating_calls.try_push(call_index).map_err(|_| Error::<T>::TooManyCalls)?;
 				// Check user has enough funds and send it to the Lottery account.
 				T::Currency::transfer(caller, &Self::account_id(), config.price, KeepAlive)?;
```

Comment:
Please check when the transfer here fails, that this is still a noop, and that participating calls push does not somehow modify storage.

Reply:
I added some more tests after discovering the `assert_storage_noop` macro :sparkles: